src/main/java/org/folio/fqm/repository/ResultSetRepository.java

```diff
@@ -42,35 +42,45 @@ public class ResultSetRepository {
 
   public List<Map<String, Object>> getResultSet(UUID entityTypeId,
                                                 List<String> fields,
-                                                List<UUID> ids) {
+                                                List<List<String>> ids) {
     if (CollectionUtils.isEmpty(fields)) {
       log.info("List of fields to retrieve is empty. Returning empty results list.");
       return List.of();
     }
+    List<String[]> idsArray = new ArrayList<>();
+    ids.forEach(id -> idsArray.add(id.toArray(new String[0])));
     EntityType entityType = getEntityType(entityTypeId);
     var fieldsToSelect = getSqlFields(entityType, fields);
-    String idValueGetter = entityType
-      .getColumns()
-      .stream()
-      .filter(col -> ID_FIELD_NAME.equals(col.getName()))
-      .map(col -> col.getValueGetter() != null ? col.getValueGetter() : col.getName())
-      .findFirst()
-      .orElseThrow(() -> new ColumnNotFoundException(entityType.getName(), ID_FIELD_NAME));
-
+    Field<String[]> idValueGetter = IdColumnUtils.getResultIdValueGetter(entityType);
     var result = jooqContext.select(fieldsToSelect)
       .from(entityType.getFromClause())
-      .where(field(idValueGetter).in(ids))
+      .where(idValueGetter.in(idsArray))
       .fetch();
     return recordToMap(result);
   }
 
-  public List<Map<String, Object>> getResultSet(UUID entityTypeId, Fql fql, List<String> fields, UUID afterId, int limit) {
+  public List<Map<String, Object>> getResultSet(UUID entityTypeId, Fql fql, List<String> fields, List<String> afterId, int limit) {
     if (CollectionUtils.isEmpty(fields)) {
       log.info("List of fields to retrieve is empty. Returning empty results list.");
       return List.of();
     }
-    Condition afterIdCondition = afterId != null ? field(ID_FIELD_NAME).greaterThan(afterId) : DSL.noCondition();
     EntityType entityType = getEntityType(entityTypeId);
+    List<String> idColumnNames = IdColumnUtils.getIdColumnNames(entityType);
+    Field<String[]> idValueGetter = IdColumnUtils.getResultIdValueGetter(entityType);
+    Condition afterIdCondition;
+    if (afterId != null) {
+      String[] afterIdArray = afterId.toArray(new String[0]);
+      afterIdCondition = field(idValueGetter).greaterThan(afterIdArray);
+    } else {
+      afterIdCondition = DSL.noCondition();
+    }
+    // Make sure idColumns are included in results
+    idColumnNames.forEach(idColumn -> {
+      if (!fields.contains(idColumn)) {
+        fields.add(idColumn);
+      }
+    });
```

Comment:
Nit: I'd either use a stream for this (so that you can do `.filter()` instead of the `if`) or do it with an old school for-each loop (`for (var idColumn : idColumnNames) {`), just to stick with a single idiom. Personally, I'd probably go old school for this one, since it's so simple and matches the style of the rest of the method